explore/src/main/scala/explore/proposal/ProgramUsersTable.scala

```diff
@@ -70,29 +76,78 @@ object ProgramUsersTable:
   ): ColumnDef.Single.WithTableMeta[ProgramUserWithRole, V, TableMeta] =
     ColDef(id, accessor, columnNames(id))
 
+  val partnerLinkOptions: List[PartnerLink] =
+    PartnerLink.HasNonPartner :: Enumerated[Partner].all.map { p =>
+      PartnerLink.HasPartner(p)
+    }
+
+  def partnerItem(pl: PartnerLink): VdomNode = pl match {
+    case PartnerLink.HasPartner(p) =>
+      <.div(
+        ExploreStyles.PartnerFlagItem,
+        <.img(
+          ^.src := PartnerFlags.smallFlag(p),
+          ^.alt := s"${p.shortName} Flag",
+          ExploreStyles.PartnerSplitFlag
+        ),
+        p match {
+          case Partner.UH => "U of Hawaii"
+          case p          => p.shortName
+        }
+      )
+    case PartnerLink.HasNonPartner => <.div("No Partner")
+    case _                         => "Unspecified Partner"
+  }
+
+  def partnerSelector(value: Option[PartnerLink], set: Option[PartnerLink] => Callback): VdomNode =
+    FormDropdownOptional(
+      id = "user-partner-selector".refined,
+      placeholder = "Select a partner",
+      options = partnerLinkOptions.map { pl =>
+        new SelectItem[PartnerLink](value = pl, label = pl.toString)
+      },
+      clazz = ExploreStyles.PartnerSelector,
+      showClear = true,
+      itemTemplate = pl => partnerItem(pl.value),
+      valueTemplate = pl => partnerItem(pl.value),
+      emptyMessageTemplate = "No Selection",
+      value = value,
+      onChange = set
+    )
+
+  def partnerLinkLens(userId: User.Id) =
+    each[List[ProgramUserWithRole], ProgramUserWithRole]
+      .filter(_.user.id === userId)
+      .andThen(ProgramUserWithRole.partnerLink)
+
   private def columns(
-    ctx: AppContext[IO]
+    pid:   Program.Id,
+    users: View[List[ProgramUserWithRole]],
+    ctx:   AppContext[IO]
   ): List[ColumnDef.WithTableMeta[ProgramUserWithRole, ?, TableMeta]] =
     List(
       column(NameColumnId, _.name),
       ColDef(
         PartnerColumnId,
-        _.partnerLink.flatMap {
-          case PartnerLink.HasPartner(partner)   => Some(partner)
-          case PartnerLink.HasNonPartner         => None
-          case PartnerLink.HasUnspecifiedPartner => None
-        },
-        "",
+        identity,
```

Comment:
This breaks sorting. We probably want what you have further down as `pl` as the column value.

Then, in the cell render function, you can still access the row as `cell.row.original` to get the `usersView`.